cosign.py

```diff
@@ -98,7 +98,9 @@ def sign_image(
     on_exist: OnExist|str=OnExist.APPEND,
     signature_image_reference: str=None,
     oci_client: oc.Client=None,
-):
+    manifest: om.OciImageManifest=None,
+    dry_run: bool=False,
```

Comment:
hmmm :-/
How about passing-in a mocked oci_client, instead? As an alternative, we might factor-out the signature-creation-part (however, I think mocking might actually not be a bad approach here)